lib/src/repository/check_repository.dart

```diff
@@ -58,47 +64,86 @@ Future<Repository?> checkRepository(PackageContext context) async {
         failVerification('Repository has no `pubspec.yaml`.');
         return result();
       }
-      final pubspecPath = p.join(url.path, 'pubspec.yaml');
-      if (!pubspecFiles.contains(pubspecPath)) {
-        // TODO: detect if any of them is matching
-        failVerification(
-            'Repository has no `pubspec.yaml` in location: `${url.path}/`.');
-        return result();
+
+      Future<_PubspecMatch> matchRepoPubspecYaml(String path) async {
+        // checkout the pubspec.yaml at the assumed location
+        final content = await repo.showStringContent(
+          branch!,
+          path,
+          maxOutputBytes: _maxPubspecBytes,
+        );
+        final gitPubspec = Pubspec.parseYaml(content);
```

Comment:
We should probably handle `FormatException` here, so that we don't fail because some invalid pubspec.yaml exist in the repository.

Reply:
I'd start with the current strict approach, and see how it goes. I'll add a TODO to revisit it later.